### PR TITLE
Support the full `attach` options parameter in the node version too.

### PR DESCRIPTION
**Pull request by lapo-luchini**
As requested in #1079.

### CODE REVIEW

**Comment by lapo-luchini** [comment]
(I'm not sure about travis, but it didn't break `make test-browser-local`)

**Comment by martinbooth** [comment]
Hey. What is `filename` in https://github.com/visionmedia/superagent/blob/b179a6b3766b8f322b1d3bd91469c4755e190dc9/lib/node/index.js#L185 ? Doesn't seem to be defined anywhere is it?

File: lib/client.js [hunk 1]
```diff
@@ -660,13 +660,13 @@ Request.prototype.query = function(val){
...
- * @param {String} filename
+ * @param {String|Object} options
  * @return {Request} for chaining
  * @api public
  */
 
-Request.prototype.attach = function(field, file, filename){
-  this._getFormData().append(field, file, filename || file.name);
+Request.prototype.attach = function(field, file, options){
+  this._getFormData().append(field, file, options || file.name);
```
Comment (lapo-luchini) [inline lib/client.js]: No, as the [third parameter is an object](https://github.com/form-data/form-data/blob/19748778e2cd05e2ddc95409fcbbf0461a868775/lib/form_data.js#L44), even though a string is accepted.

**Comment by lapo-luchini** [comment]
@martinbooth you're right, that should be a `o.filename`.

**Pull request merged**